tests/integration/sparse_pipeline/test_sparse_pipeline.py

```diff
@@ -34,23 +34,23 @@ def encode(self, data: Any, *args, **kwargs) -> Any:
 
 
 class DummyCSRSparseIndexer(BaseSparseVectorIndexer):
-
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.keys = []
 
-    def add(self, keys: Iterable[str], vectors: 'scipy.sparse.coo_matrix', *args, **kwargs) -> None:
+    def add(
+        self, keys: Iterable[str], vectors: 'scipy.sparse.coo_matrix', *args, **kwargs
+    ) -> None:
         print(type(vectors))
         assert isinstance(vectors, sparse.coo_matrix)
         self.keys.extend(keys)
 
-
-    def query(
-        self, vectors: 'scipy.sparse.coo_matrix', top_k: int, *args, **kwargs
-    ):
+    def query(self, vectors: 'scipy.sparse.coo_matrix', top_k: int, *args, **kwargs):
         assert isinstance(vectors, sparse.coo_matrix)
         distances = [item for item in range(0, min(top_k, len(self.keys)))]
-        return np.array(self.keys[:top_k]), np.array(distances)
+        return np.array(self.keys[:top_k]), np.array(
```

Comment:
I think this was a bug, distances needs to be an array of arrays (for each vector I want the closest distances)

Reply:
query is not done in batches